### PR TITLE
[META-88] Command interface update

### PR DESCRIPTION
**Pull request by DogukanGun**
Description of Issue
[Linear Task](https://linear.app/la-famiglia-project/issue/META-88/style-update-for-command-interface)

How I implemented
Row type now has restriction param and arrow down and up buttons are now working

### CODE REVIEW

File: src/components/side-effects/CommandInterface/index.tsx [hunk 1]
```diff
@@ -75,7 +108,26 @@ const CommandInterface: React.FC = () => {
...
+  useEffect(() => {
+    if (projects) {
+      setRecommendationList(
+        defaultList.map((item) => {
+          const newItem = { ...item }
+          if (newItem.commandInterfaceOptions.includes('mailto:')) {
+            const project = projects.filter((projectItem) => projectItem.id === id)[0]
+            newItem.commandInterfaceOptions = emailTemplate(
+              project.associatedPerson?.email ?? '',
+              project.associatedPerson?.name ?? '',
```
Comment (kigadye) [inline src/components/side-effects/CommandInterface/index.tsx:122]:
```suggestion
              project?.associatedPerson?.email ?? '',
              project?.associatedPerson?.name ?? '',
```

**Review by kigadye — CHANGES_REQUESTED** [review]
When I checkout your branch locally and run it I get the following error:
Screenshot 2023-06-20 at 00 02 04

When I fix that (see comment below), it works, but there seems to be an infinite loop somewhere:
Screenshot 2023-06-20 at 00 01 40

**Review by kigadye — CHANGES_REQUESTED** [review]
Arrow key navigation works, but there are some weird bugs:
- I can't reload the page using Cmd + R anymore
- I can't use Cmd + Backspace to delete the entire line in the CI
- When I go up with the arrow keys, it seems like it goes out of the command interface and keeps counting up, because if I go down again I need to press down the same number of times I went up "out of the CI"
- Sometimes, items do the actions of another one (e.g. settings does logout, logout goes to docs)
- Send email doesn't populate the founder email and name
Screenshot 2023-06-22 at 00 33 06

**Comment by DogukanGun** [comment]
> Arrow key navigation works, but there are some weird bugs:
> 
> * I can't reload the page using Cmd + R anymore
> * I can't use Cmd + Backspace to delete the entire line in the CI
> * When I go up with the arrow keys, it seems like it goes out of the command interface and keeps counting up, because if I go down again I need to press down the same number of times I went up "out of the CI"
> * Sometimes, items do the actions of another one (e.g. settings does logout, logout goes to docs)
> * Send email doesn't populate the founder email and name
> 
> Screenshot 2023-06-22 at 00 33 06
 
I checked the email issue and it is happening because email is not exist in that project. I tried other projects and some of them has an email. Other issues are fixed.
Screenshot 2023-06-22 at 10 18 13

**Review by kigadye — CHANGES_REQUESTED** [review]
Most issues are fixed, except when I use the arrow keys to navigate through items and press enter, they still do the wrong actions (e.g. settings does logout, logout goes to docs).

Also, regarding send email for projects that don't have an email, either show a message in the command interface that there is no founder data for this project when the user clicks send email, or just hide the send email option from the CI on those pages.

**Comment by kigadye** [comment]
Another bug: the CI opens when clicking esc

**Comment by DogukanGun** [comment]
> Another bug: the CI opens when clicking esc

I will check this. I could not catch the other issue (settings does logout, logout goes to docs) even I tried multiple times. Could you just check your env or try one more time with another browser. Or may be you can share in which page you faced with this issue?

**Comment by kigadye** [comment]
now the command interface doesn't open at all when I press Cmd/ Ctrl + K

**Comment by kigadye** [comment]
now the command interface doesn't open at all when I press Cmd/ Ctrl + K

**Comment by DogukanGun** [comment]
> now the command interface doesn't open at all when I press Cmd/ Ctrl + K

fixed

**Review by kigadye — CHANGES_REQUESTED** [review]
All of the issues I mentioned above are fixed, but navigating to project details doesn't work

**Comment by DogukanGun** [comment]
> All of the issues I mentioned above are fixed, but navigating to project details doesn't work

Could you check this video, it is working on me.
https://www.loom.com/share/bd619eee508a425babdb9b0d66fb0199?sid=1b164fb6-4bb7-460d-afea-d8dd94f0a2f5

**Review by kigadye — CHANGES_REQUESTED** [review]
navigation should work with keyboard only

**Review by kigadye — APPROVED** [review]

**Review by ArnoClaude — APPROVED** [review]
Good job @DogukanGun

**Pull request merged**